plugins/woocommerce/includes/class-wc-shipping.php

```diff
@@ -358,6 +358,27 @@ public function calculate_shipping_for_package( $package = array(), $package_key
 				}
 			}
 
+			// Hide shipping rates when free shipping is available.
+			if ( 'yes' === get_option( 'woocommerce_shipping_hide_rates_when_free', 'no' ) ) {
+				$free_shipping = array();
+				$local_pickup  = array();
+
+				foreach ( $package['rates'] as $rate ) {
+					if ( 'free_shipping' === $rate->method_id ) {
+						$free_shipping[] = $rate;
+						continue;
+					}
+
+					if ( 'local_pickup' === $rate->method_id ) {
```

Comment:
I believe that `local_pickup` is the legacy option that is only available if a store was using it, the new version is named `pickup_location`. I'm wondering if we should include both for backwards compatibility.

This works:

```
					if ( in_array( $rate->method_id, array( 'local_pickup', 'pickup_location' ) ) ) {
						$local_pickup[] = $rate;
					}
```

The way I tested with the legacy method was to find [the `is_legacy_local_pickup_active` method](https://github.com/woocommerce/woocommerce/blob/8f39c7d6827622e7434e4820712ba39beb866dc2/plugins/woocommerce/src/Blocks/Shipping/ShippingController.php#L551) and set it to return `true`. I was then able to add the method to a shipping zone.

Reply:
Thanks for mentioning this! My local env isn't working well because I only see the legacy way of adding it, the new one shows an empty page. But after setting up a new env site, I've noticed what you're referring to.

Addressed in 99e31c3.